### DISCUSSION TITLE
Starting Directory for the Filebrowser Mode and Recursive Searching

### DISCUSSION THREAD

**Discussion by protean-onion** [davatorium/rofi#1815] [(Possible) Issues]
Hello,

I'm trying to use rofi as the Spotlight feature on a Mac. There are some issues that I am having.

First, specifying a starting directory for the "directory" configuration option does not result in rofi's starting in that directory. rofi starts in the directory in which it was previously.

Second, I'd like it if rofi could recursively search for files through subdirectories of the current directory. The Spotlight feature on a Mac is really helpful in this regard, as you can open any file with the press of just a few buttons. In rofi, the user is required to navigate through the file system and then search for the file in the directory in which it is located. A Spotlight-like feature on rofi would be great.

Thanks for your time. I look forward to your response.

Warm Regards,
Aditya

**Comment by DaveDavenport** [comment]
> First, specifying a starting directory for the "directory" configuration option does not result in rofi's starting in that directory. rofi starts in the directory in which it was previously.

Are you running a version new enough to implement this feature? 
What version do you run?

**Reply by protean-onion** [reply]
The latest one, but I've decided against using rofi. I was having too many dependency issues, and I don't know enough to deal with all of that.

Thanks for taking the time to respond to me. If you would like me to still share the details regarding the problem so you can improve it, please let me know. In general, though, I think I was having some problem with `libxkbcommons`.

**Reply by DaveDavenport** [reply]
The latest as in git or latest as in release?

Rofi uses very common libraries that are available on most linux distros (you have not mentioned what distro you use) .  Especially a library like libxkbcommon is so common, I would find it strange if it is missing. Not sure there is anything I can do from my side to 'improve' this.
Again this is me guessing as you shared  very very little details.

**Comment by DaveDavenport** [comment]
>Second, I'd like it if rofi could recursively search for files through subdirectories of the current directory. The Spotlight feature on a Mac is really helpful in this regard, as you can open any file with the press of just a few buttons. In rofi, the user is required to navigate through the file system and then search for the file in the directory in which it is located. A Spotlight-like feature on rofi would be great.

This would be slow, if a feature like spotlight should be implemented, it should be combined with something like tracker. (https://wiki.gnome.org/Projects/Tracker)

**Reply by protean-onion** [reply]
Thanks! I'll check it out, though I've committed to working with `dmenu` and `lf`.
